### PR TITLE
feat: Add new logging api (#9631)

### PR DESCRIPTION
**Pull request by timo0**
fixes https://github.com/hashgraph/hedera-services/issues/9631
fixes https://github.com/hashgraph/hedera-services/issues/9573
fixes https://github.com/hashgraph/hedera-services/issues/9000
fixes https://github.com/hashgraph/hedera-services/issues/9001

### CODE REVIEW

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Loggers.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Loggers.java:43]:
```suggestion
    public static Logger getLogger(@NonNull final String name) {
```

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Loggers.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Loggers.java:59]:
```suggestion
    public static Logger getLogger(@NonNull final Class<?> clazz) {
```

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/extensions/handler/AbstractSyncedHandler.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/extensions/handler/AbstractSyncedHandler.java:60]:
```suggestion
    public final void accept(@NonNull final LogEvent event) {
```

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/LineBasedFormat.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/LineBasedFormat.java:57]: in that case we can directly return from the method

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/LineBasedFormat.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/LineBasedFormat.java:55]: in that case we can directly return from the method

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/LoggingLevelConfig.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/LoggingLevelConfig.java:45]: Do we still need that class?

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandler.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandler.java:24]: Javadoc is missing

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandler.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandler.java:26]: javadoc + NonNull

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandler.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandler.java:31]: NonNull

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandlerFactory.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandlerFactory.java:23]: Javadoc

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandlerFactory.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/console/ConsoleHandlerFactory.java:26]: NonNull

File: platform-sdk/swirlds-logging/src/main/java/module-info.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/java/module-info.java:16]: why?

File: platform-sdk/swirlds-logging/src/main/resources/META-INF/services/com.swirlds.logging.api.extensions.handler.LogHandlerFactory
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/main/resources/META-INF/services/com.swirlds.logging.api.extensions.handler.LogHandlerFactory:1]: We now can use Google Autoservice to generate that file. 
Example for config-impl:

annotation processor def in gradle file: https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/swirlds-config-impl/build.gradle.kts
annotation usage: https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConfigurationBuilderFactoryImpl.java

File: platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/HandlerLoggingLevelConfigStressTest.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/HandlerLoggingLevelConfigStressTest.java:38]: Instead of doing it like that I would prefer a test with params. random values in tests are always bad

File: platform-sdk/swirlds-logging/src/main/java/module-info.java
Comment (timo0) [reply platform-sdk/swirlds-logging/src/main/java/module-info.java:16]: Not needed anymore

**Review by cody-littley — DISMISSED** [review]
Giving hashgraph team signoff for the stuff owned by the hashgraph team (aka zero lines in this PR 🙃). I think we are still marked as code owners because of the old legacy logging stuff that used to be in this location.

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java:91]:
```suggestion
    public HandlerLoggingLevelConfig(@NonNull final Configuration configuration, @Nullable String name) {
```

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java:103]:
```suggestion
    public HandlerLoggingLevelConfig(@NonNull final Configuration configuration) {
```

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java:113]: is the convention @NotNull final / final @NotNull or we don't care?

File: platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/LoggingSystemTest.java
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/LoggingSystemTest.java:394]:
```suggestion
            "Test that checks if simple log calls are forwarded correctly will all information to the configured handler")
```

File: platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/LoggingSystemTest.java
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/LoggingSystemTest.java:453]:
```suggestion
    @DisplayName("Test that accept passes events to the configured handler")
```

File: platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/LoggingSystemTest.java
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/LoggingSystemTest.java:488]:
```suggestion
            "That that checks if complex log calls are forwarded correctly with all informations to the configured handler")
```
```suggestion
            "Test that checks if complex log calls are forwarded correctly with all information to the configured handler")
```

File: platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/LoggingSystemTest.java
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/LoggingSystemTest.java:569]:
```suggestion
    @DisplayName("Test that accept passes complex log calls correctly with all informations to the configured handler")
```

**Review by mxtartaglia-sl — COMMENTED** [review]
Leaving some comments, so we can talk about it more on a call. Sorry for those that are questions or comments fixes; guessed that at least I could give a hand there since there is not much to add from the code side.

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Loggers.java
Comment (hendrikebbers) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Loggers.java:27]: that is a typo ;) should be: "To not mix it up with other logger factories we decided to use Loggers as a name that is not used in any other logging framework."

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/StackTracePrinter.java
Comment (hendrikebbers) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/StackTracePrinter.java:104]: yes, that is often the problem with lazy Hendrik ;)

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/event/ReuseableLogEventFactory.java
Comment (hendrikebbers) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/event/ReuseableLogEventFactory.java]: yes we should.
The factory is currently not used in production. in future (next step) we might switch to that factory and need to take care of that.

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java
Comment (hendrikebbers) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java:240]: 🙈 YES, I would do it like that: create an issue for the bug and fix it once the PR is merged.

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java
Comment (hendrikebbers) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java:112]: Had the same question some time ago but missed to check it. We do not have it defined in our code style so far but it would make sense to define best practices (and maybe even have our code formatter check that). Can you create an issue for that?

Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java:112]: Created issue: https://github.com/hashgraph/hedera-services/issues/10380

File: platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Logger.java
Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/Logger.java:25]: created issue:https://github.com/hashgraph/hedera-services/issues/10389

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]
Approving again. (Note that I didn't actually review anything in the logging module, I consider me being a code owner of that module to be a misconfiguration.)

**Review by mxtartaglia-sl — DISMISSED** [review]

**Review by hendrikebbers — APPROVED** [review]

**Review by hendrikebbers — DISMISSED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Comment by cody-littley** [comment]
@timo0 this is marked as milestone 0.46. For a feature like this, I'm assuming we don't plan on back porting. Current develop is at 0.47.

**Comment by timo0** [comment]
@cody-littley yes this PR is open too long :)

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by hendrikebbers — APPROVED** [review]

**Pull request merged**
